framework/src/wifi_manager/wifi_manager.c

```diff
@@ -0,0 +1,477 @@
+	} else {
+		dhcpc_close(g_dhcp_handle);
+		g_dhcp_handle = NULL;
+	}
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+static wifi_utils_result start_dhcp_server(void)
+{
+	struct in_addr in = {.s_addr = 0x012fa8c0};
+	netlib_set_ipv4addr(CTRL_IFNAME, &in);
+	in.s_addr = 0x00ffffff;
+	netlib_set_ipv4netmask(CTRL_IFNAME, &in);
+	in.s_addr = 0x012fa8c0;
+	netlib_set_dripv4addr(CTRL_IFNAME, &in);
+	
+	if (dhcpd_start(CTRL_IFNAME, g_new_sta_join) < 0) {
+		ndbg("DHCP Server - started fail\n");
+		return WIFI_UTILS_FAIL;
+	}
+
+	nvdbg("DHCP Server - started success\n");
+	return WIFI_UTILS_SUCCESS;
+}
+
+static wifi_utils_result stop_dhcp_server(void)
+{
+	struct in_addr in = {.s_addr = INADDR_NONE};
+	netlib_set_ipv4addr(CTRL_IFNAME, &in);
+	netlib_set_ipv4netmask(CTRL_IFNAME, &in);
+	netlib_set_dripv4addr(CTRL_IFNAME, &in);
+	
+	dhcpd_stop();
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+static void wifi_status_set(connect_status status)
+{
+#ifdef CONFIG_DEBUG_NET_INFO
+	static char connect_status_str[5][25] =
+	{
+			// STA mode
+			"AP_DISCONNECTED",
+			"AP_CONNECTED",
+
+			// SOFTAP mode
+			"CLIENT_CONNECTED",
+			"CLIENT_DISCONNECTED"
+	};
+#endif
+
+	if(g_manager_info.status != status)
+	{
+		nvdbg("Wifi Network Status Changes from [%s] to [%s]",
+				connect_status_str[g_manager_info.status], connect_status_str[status]);
+		g_manager_info.status = status;
+	}
+}
+
+static void ap_connect_event_func(void)
+{
+	wifi_manager_cb *wifi_cb = g_manager_info.wmcb;
+	if(g_manager_info.mode == STA_MODE)
+	{
+		nvdbg("WIFI CONNECTED AP - STA MODE");
+		wifi_status_set(AP_CONNECTED);
+
+		/* Perform DHCP client */
+		if (start_dhcp_client() != WIFI_UTILS_SUCCESS) {
+			ndbg("DHCP client start failed\n");
+			return ;
+		}
+		if (stop_dhcp_client() != WIFI_UTILS_SUCCESS) {
+			ndbg("DHCP client stop failed\n");
+		}
+
+		if (wifi_cb != NULL && wifi_cb->sta_connected) {
+			wifi_cb->sta_connected();
+		}
+		else {
+			ndbg("Callback wifimanager ap_connect failed\n");
+		}
+	}
+	else if (g_manager_info.mode == SOFT_AP_MODE)
+	{
+		nvdbg("CONNECTED FROM CLIENT - SOFT AP MODE");
+		wifi_status_set(CLIENT_CONNECTED);
+		
+		/* No callbacks here, in case new client joins, we invoke callback
+		 * from DHCP server instead
+		 */		 
+	}
+#ifdef CONFIG_ENABLE_IOTIVITY
+	__tizenrt_manual_linkset("gen");
+#endif
+	/* TODO: Import files from source 
+	 * sendStatusTrigger (TRIGGER_NETWORK_CHANGED, DAWIT_NW_CHANGED_UP);
+	 */
+}
+
+static void ap_disconnect_event_func(void)
+{
+	wifi_manager_cb *wifi_cb = g_manager_info.wmcb;	
+	if(g_manager_info.mode == STA_MODE)
+	{
+		nvdbg("WIFI DISCONNECTED AP - STA MODE");
+		strncpy(g_manager_info.ssid, "", 32);
+		strcpy(g_manager_info.ip4_address, "");
+		g_manager_info.rssi = 0;
+		wifi_status_set(AP_DISCONNECTED);
+
+		if (wifi_cb != NULL && wifi_cb->sta_disconnected) {
+			wifi_cb->sta_disconnected();
+		}
+		else {
+			ndbg("Callback wifimanager ap_disconnected failed\n");
+		}
+	}
+	else if (g_manager_info.mode == SOFT_AP_MODE)
```

Comment:
same as previous, and there are many same violations. Please fix them.

Reply:
I did go through fomatter.sh